client/src/related/relatedList.jsx

```diff
@@ -1,69 +1,80 @@
 import React from 'react';
 import ProductCard from './productCard.jsx';
+import ComparisonModal from './comparisonModal.jsx';
 import axios from 'axios';
 
 class RelatedList extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      products: [
+      relatedProducts: [
         {
-          'id': 11,
-          'name': 'Air Minis 250',
-          'category': 'Basketball Shoes',
+          'id': 2,
+          'name': 'Product Name',
+          'category': 'Product Category',
           'original_price': '0',
-          'image': 'urlplaceholder/style_1_photo_number.jpg',
-          'avgReview': 4.3,
-          'features': [
-            {
-              'feature': 'Sole',
-              'value': 'Rubber'
-            },
-            {
-              'feature': 'Material',
-              'value': 'FullControlSkin'
-            }
-          ]
+          'image': 'https://static.thenounproject.com/png/1103191-200.png',
+          'avgReview': 0,
         }
-      ]
+      ],
+      showModal: false,
+      currentProd: {},
+      comparisonProd: {}
     };
+    this.onStarClick = this.onStarClick.bind(this);
+    this.closeModal = this.closeModal.bind(this);
   }
 
-  refreshItemList() {
+  componentDidMount() {
```

Comment:
This is just a question(I don't have the answer) but is it better to only set state once in component did mount? Like just make one API call, then the other API call in the 'then' of the first and then set the state at the end?  Every time I have tried to set state more than once in the same function, React yells at me so I just assumed at first glance that it wouldn't work, but that is obviously not true.

Reply:
I think the reason it works is because the second call doesn't rely on the first call at all. (I think of using 'then' for when I need to pass data from the first call in to the second call.) I did just look though and it looks like if I "promise.all"ed them, they could potentially execute at the same time instead of asynchronously, so that could be a potential performance improvement for the future.